## **Import Libraries**

In [0]:
# Import pandas library
import pandas as pd

# Import the Numpy library
import numpy as np

## **Load datasets**

### **Cells Geo Dataset**

In [3]:
# Load cells geo dataset
cells_geo = pd.read_csv('sample_data/cells_geo.csv', sep=';', index_col=0)
cells_geo.head()

VILLES      STATUS LOCALISATION  ... AREA     CELL_ID   SITE_CODE
0   ADJAME  In Service      ABIDJAN  ...  NaN  fbb2e75af3  0bb492ef17
1   ADJAME  In Service      ABIDJAN  ...  NaN  1502501d78  0bb492ef17
2   ADJAME  In Service      ABIDJAN  ...  NaN  0c2e182a0c  0bb492ef17
3  ASSINIE  In Service    INTERIEUR  ...  NaN  69bdfc69fd  8a1e6ce31f
4  ASSINIE  In Service    INTERIEUR  ...  NaN  7a1029d2ab  8a1e6ce31f

[5 rows x 11 columns]

### **Cells Geo Description Dataset**

In [5]:
# Load cells geo description dataset
cells_geo_description = pd.read_excel('sample_data/cells_geo_description.xlsx')
cells_geo_description

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [6]:
# Check the villes whose status is in service
vill_in_serv = cells_geo[cells_geo['STATUS'] == 'In Service']
vill_in_serv
vill_in_serv['VILLES'].value_counts().head()

COCODY         354
YOPOUGON       289
ABOBO          250
MARCORY        141
TREICHVILLE    132
Name: VILLES, dtype: int64

*#cocody ville has the highest number of cells still in service with 354 sites,*

In [7]:
# dropping null values from cells_geo
cells_geo.dropna(inplace=True)
cells_geo.head()

VILLES      STATUS LOCALISATION  ...           AREA     CELL_ID   SITE_CODE
23  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  f740406877  0f34c36ac5
24  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  959e54c8f7  0f34c36ac5
25  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  56b7e7000c  0f34c36ac5
26  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  bc51c17a6f  0f34c36ac5
27  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  e460975c9f  0f34c36ac5

[5 rows x 11 columns]

In [10]:
# Rename Site code column to Site ID
cells_geo.rename(columns={'SITE_CODE': 'SITE_ID'}, inplace=True)
cells_geo.head()

VILLES      STATUS LOCALISATION  ...           AREA     CELL_ID     SITE_ID
23  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  f740406877  0f34c36ac5
24  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  959e54c8f7  0f34c36ac5
25  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  56b7e7000c  0f34c36ac5
26  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  bc51c17a6f  0f34c36ac5
27  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  e460975c9f  0f34c36ac5

[5 rows x 11 columns]

### **CDR Description Dataset**

In [11]:
# Load CDR Dataset
CDR = pd.read_excel('sample_data/CDR_description.xlsx')
CDR

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

### **Telcom Dataset 1**

In [12]:
#Load the first dataset; for day 1
day1 = pd.read_csv('sample_data/Telcom_dataset.csv')
day1.head(10)

PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv
5    data      0  2012-05-06 23:07:26.0  ...     OTHER  /lrUXVGVcN  /lrUXVGVcN
6   Voice      0  2012-05-06 23:08:02.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
7   Voice      0  2012-05-06 23:08:03.0  ...     19e2e  09UjmmY1Ds  09UjmmY1Ds
8   Voice      0  2012-05-06 23:08:05.0  ...     19e2e  0DeimCpZOc  0DeimCpZOc
9     sms      0  2012-05-06 23:08:43.0  ...     19e2e  01MDiY5F89  01MDiY5F89

[10 rows x 10 columns]

### **Telcom Dataset 2**

In [14]:
#Load the second dataset; for day 2
day2 = pd.read_csv('sample_data/Telcom_dataset2.csv')
day2.head(10)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO
5     sms      0  2012-05-07 23:05:41.0  ...     19e2e  /MMplBlWjh  /MMplBlWjh
6   Voice     41  2012-05-07 23:06:14.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
7   Voice     11  2012-05-07 23:06:28.0  ...     19e2e  +gySf02tKW  +gySf02tKW
8     sms      0  2012-05-07 23:06:35.0  ...     OTHER  /0FxP3az1x  /0FxP3az1x
9     sms      0  2012-05-07 23:06:50.0  ...     19e2e  0ecjmwPX21  0ecjmwPX21

[10 rows x 10 columns]

### **Telcom Dataset 3**

In [15]:
#Load the third dataset; for day 3
day3 = pd.read_csv('sample_data/Telcom_dataset3.csv')
day3.head(10)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA
5   Voice     10  2012-05-08 23:03:59.0  ...     19e2e  +e+xcimwre  +e+xcimwre
6     sms      0  2012-05-08 23:06:06.0  ...     19e2e  +R5ccmtyJf  +R5ccmtyJf
7   Voice     10  2012-05-08 23:06:55.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
8     sms     25  2012-05-08 23:07:09.0  ...     19e2e  13O1+JEun3  13O1+JEun3
9     sms      0  2012-05-08 23:07:38.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[10 rows x 10 columns]

## **Clean the data**

In [16]:
#Rename the columns and making them similar
#
columns = ['PRODUCT', 'VALUE', 'DATE_TIME', 'CELL_ON_SITE', 'DW_NUMBER_INT', 
           'DW_B_NUMBER_INT', 'COUNTRY_A', 'COUNTRY_B', 'CELL_ID', 'SITE_ID']
day1.columns = columns
day1

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0      Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1        sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2        sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3      Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv
...      ...    ...                    ...  ...       ...         ...         ...
4996     sms    100  2012-05-07 00:01:54.0  ...     OTHER  0c2371d9bc  1e751d08a4
4997   Voice      1  2012-05-07 00:01:54.0  ...     19e2e  5fb791cf56  65efe2bc62
4998   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  7773627c12  9ef9616350
4999   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  3f4af49d6c  969b8d0e39
5000   Voice      7  2012-05-07 00:01:54.0  ...     19e2e  6f4d9b77a5  ebfc58af70

[5001 rows x 10 columns]

In [18]:
# Telcomdataset 2
day2.columns = columns
day2

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0        sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1      Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2        sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3      Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4        sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO
...      ...    ...                    ...  ...       ...         ...         ...
4996     sms      0  2012-05-08 00:01:53.0  ...     19e2e  896459aace  b7d00e59e4
4997     sms     25  2012-05-08 00:01:53.0  ...     19e2e  7e90b35166  211015b14d
4998     sms      0  2012-05-08 00:01:53.0  ...     19e2e  fe70e431df  c7ea989264
4999    data      0  2012-05-08 00:01:53.0  ...     OTHER  ffa6759bb2         NaN
5000   Voice      0  2012-05-08 00:01:53.0  ...     19e2e  8630058aca  b47a998a3e

[5001 rows x 10 columns]

In [19]:
# Telcomdataset 3
day3.columns = columns
day3

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0      Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1        sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2      Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3        sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA
...      ...    ...                    ...  ...       ...         ...         ...
4996   Voice      0  2012-05-09 00:01:43.0  ...     19e2e  2aa04dcab7  f7d63a29e5
4997   Voice     62  2012-05-09 00:01:43.0  ...     19e2e  7f1f2f889f  b4ddb9e93c
4998   Voice     32  2012-05-09 00:01:43.0  ...     19e2e  a4731e780e  d68260ef50
4999   Voice     30  2012-05-09 00:01:43.0  ...     19e2e  98ed726da2  b4130b54ab
5000   Voice     60  2012-05-09 00:01:43.0  ...     19e2e  1c11a2fcbb  6d70dae094

[5001 rows x 10 columns]

### **Concatenating the three datasets**

In [20]:
#Join the three datasets
df_joined = pd.concat([day1, day2, day3], ignore_index = True)
df_joined.head()

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [21]:
# Drop the columns that are not useful
df_joined.drop(columns= ['COUNTRY_A', 'COUNTRY_B', 'CELL_ON_SITE'], inplace=True)
df_joined.head()

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0   Voice      0  ...  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  ...  /+cKZKIp41  /+cKZKIp41
2     sms     25  ...  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  ...  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  ...  051KhYDCpv  051KhYDCpv

[5 rows x 7 columns]

#### **Splitting the date_time column to date and time**

In [22]:
#Splitting the date_time column and creating two new columns to hold the two variables 
df_joined[['date', 'time']] = df_joined.DATE_TIME.str.split(expand=True)
df_joined.head()

PRODUCT  VALUE              DATE_TIME  ...     SITE_ID        date        time
0   Voice      0  2012-05-06 23:04:37.0  ...  /TJNe+Mmtu  2012-05-06  23:04:37.0
1     sms      0  2012-05-06 23:05:25.0  ...  /+cKZKIp41  2012-05-06  23:05:25.0
2     sms     25  2012-05-06 23:05:41.0  ...  /u0FSD+ahi  2012-05-06  23:05:41.0
3   Voice     56  2012-05-06 23:05:42.0  ...  +cNeJzsTp3  2012-05-06  23:05:42.0
4   Voice     14  2012-05-06 23:06:29.0  ...  051KhYDCpv  2012-05-06  23:06:29.0

[5 rows x 9 columns]

In [23]:
#Dropping DATE_TIME column
df_joined.drop(columns=['DATE_TIME'], inplace=True)
df_joined.head()

PRODUCT  VALUE    DW_NUMBER_INT  ...     SITE_ID        date        time
0   Voice      0  49fae048b547780  ...  /TJNe+Mmtu  2012-05-06  23:04:37.0
1     sms      0  7a54a6eaee19e65  ...  /+cKZKIp41  2012-05-06  23:05:25.0
2     sms     25  6008887b0207d82  ...  /u0FSD+ahi  2012-05-06  23:05:41.0
3   Voice     56  7370032e20f4e96  ...  +cNeJzsTp3  2012-05-06  23:05:42.0
4   Voice     14  966ba4656e38e62  ...  051KhYDCpv  2012-05-06  23:06:29.0

[5 rows x 8 columns]

#### **Inspecting data in concatenated dataset**

In [24]:
# Data in the joined dataset
# groupby product
df_data = df_joined.groupby(['time', 'PRODUCT'])[['PRODUCT']].count()
df_data.head(10)

PRODUCT
time       PRODUCT         
00:00:00.0 Voice         28
           data           4
           sms           35
00:00:01.0 Voice         77
           data          11
           sms           48
00:00:02.0 Voice         47
           data          12
           sms           44
00:00:03.0 Voice         40

In [26]:
df_data.tail(10)

PRODUCT
time       PRODUCT         
23:59:49.0 Voice          1
23:59:50.0 sms            1
23:59:51.0 sms            1
23:59:53.0 sms            2
23:59:54.0 Voice          1
23:59:55.0 sms            2
23:59:56.0 Voice          3
23:59:57.0 sms            1
23:59:58.0 Voice          1
           sms            1

In [25]:
# Data in the joined dataset
# groupby value
df_data2 = df_joined.groupby(['time', 'PRODUCT'])[['VALUE']].sum()
df_data2.head(10)

VALUE
time       PRODUCT       
00:00:00.0 Voice      666
           data         0
           sms        285
00:00:01.0 Voice     3253
           data         0
           sms        562
00:00:02.0 Voice     1359
           data         3
           sms        334
00:00:03.0 Voice     5814

In [27]:
df_data2.tail(10)

VALUE
time       PRODUCT       
23:59:49.0 Voice        0
23:59:50.0 sms         50
23:59:51.0 sms          0
23:59:53.0 sms          0
23:59:54.0 Voice      130
23:59:55.0 sms          0
23:59:56.0 Voice       29
23:59:57.0 sms          0
23:59:58.0 Voice       31
           sms          0

### **Merge the concatenated dataframe with cells_geo**

In [28]:
# Create new with merged dataframes from the concantenated above
#
merged = cells_geo.merge(df_joined, how='left', on='SITE_ID')
merged.head()

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
0  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:16.0
1  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
2  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
3  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:22.0
4  ABOBO  In Service      ABIDJAN  ...  10ae6c0752  2012-05-07  00:00:37.0

[5 rows x 18 columns]

In [29]:
# values of the merged table where time is not null
#
fmerged = merged[merged.time.notnull()]
fmerged.head()

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
0  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:16.0
1  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
2  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
3  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:22.0
4  ABOBO  In Service      ABIDJAN  ...  10ae6c0752  2012-05-07  00:00:37.0

[5 rows x 18 columns]

In [30]:
# Groupby the merged table with no null time values by time and product 

fmerged.groupby(['time', 'PRODUCT'])[['VALUE']].sum().head(10)

VALUE
time       PRODUCT         
00:00:00.0 Voice     2679.0
           sms       1333.0
00:00:01.0 Voice    13628.0
           sms       2558.0
00:00:02.0 Voice     5975.0
           sms       1479.0
00:00:03.0 Voice    24385.0
           sms       3462.0
00:00:04.0 Voice     8705.0
           sms       3078.0

**##most communication occurs during home hours**



**##voice is the most used product**

**##best time for an upgrade will be during the day**

In [32]:
#checking the maximum value to determine the most used product

fmerged[['VALUE']].max()

VALUE    6750.0
dtype: float64

In [33]:
#Checking the city with the most users of the voice product 
fmerged[fmerged['VALUE'] == 6750.0]

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
11486  COCODY  In Service      ABIDJAN  ...  3f5ac97214  2012-05-09  00:00:23.0
11499  COCODY  In Service      ABIDJAN  ...  3f5ac97214  2012-05-09  00:00:23.0
11512  COCODY  In Service      ABIDJAN  ...  3f5ac97214  2012-05-09  00:00:23.0
11525  COCODY  In Service      ABIDJAN  ...  3f5ac97214  2012-05-09  00:00:23.0
11538  COCODY  In Service      ABIDJAN  ...  3f5ac97214  2012-05-09  00:00:23.0

[5 rows x 18 columns]

In [35]:
#checking highest value of the sms product 
#
sms_val = fmerged[fmerged['PRODUCT'] == 'sms']
sms_val.head()

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
0  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:16.0
1  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
2  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
3  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:22.0
5  ABOBO  In Service      ABIDJAN  ...  10ae6c0752  2012-05-07  00:00:39.0

[5 rows x 18 columns]

In [36]:
sms_val['VALUE'].max()

100.0

In [40]:
#Getting the cities with the most users of sms
sms_val[sms_val['VALUE'] == 100.0]

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
5303     ABOBO  In Service      ABIDJAN  ...  5c7dc2f617  2012-05-07  00:00:04.0
5348     ABOBO  In Service      ABIDJAN  ...  5c7dc2f617  2012-05-07  00:00:04.0
5393     ABOBO  In Service      ABIDJAN  ...  5c7dc2f617  2012-05-07  00:00:04.0
5438     ABOBO  In Service      ABIDJAN  ...  5c7dc2f617  2012-05-07  00:00:04.0
5483     ABOBO  In Service      ABIDJAN  ...  5c7dc2f617  2012-05-07  00:00:04.0
...        ...         ...          ...  ...         ...         ...         ...
50729  MARCORY  In Service      ABIDJAN  ...  5fd5eae55a  2012-05-07  00:01:02.0
50740  MARCORY  In Service      ABIDJAN  ...  5fd5eae55a  2012-05-07  00:01:02.0
50751  MARCORY  In Service      ABIDJAN  ...  5fd5eae55a  2012-05-07  00:01:02.0
50762  MARCORY  In Service      ABIDJAN  ...  5fd5eae55a  2012-05-07  00:01:02.0
50773  MARCORY  In Service      ABIDJAN  ...  5fd5eae55a  2012-05-07  00:01:02.0

[148 rows x 18 columns]

In [41]:
#Checking the cities with most users of the network

fmerged['VILLES'].value_counts()

YOPOUGON       8540
ABOBO          6039
COCODY         3814
ADJAME         2584
KOUMASSI       2333
               ... 
SAMO              3
BROFFOUEDOU       3
TAHIRAGUHE        3
KOTOULA           2
DANANON           2
Name: VILLES, Length: 398, dtype: int64

In [0]:
##The most used cities for the three days were YOPOUGON, Abobo and Cocody respectively

#The most used city for the three days was Yopougon

In [46]:
#Checking network use outside business hours between 00:00 and 09:00hrs
obh = fmerged[(fmerged['time'] >= '00:00:00.0') & (fmerged['time'] <= '09:00:00.0')]
obh.head()

VILLES      STATUS LOCALISATION  ...   CELL_ID_y        date        time
0  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:16.0
1  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
2  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:20.0
3  ABOBO  In Service      ABIDJAN  ...  e12d87c801  2012-05-07  00:00:22.0
4  ABOBO  In Service      ABIDJAN  ...  10ae6c0752  2012-05-07  00:00:37.0

[5 rows x 18 columns]

In [43]:
obh['VILLES'].value_counts()


YOPOUGON       8540
ABOBO          6039
COCODY         3814
ADJAME         2584
KOUMASSI       2333
               ... 
SAMO              3
BROFFOUEDOU       3
TAHIRAGUHE        3
KOTOULA           2
DANANON           2
Name: VILLES, Length: 398, dtype: int64

In [0]:
##The most used cities during home hours was Yopougon, Abobo and Cocody

## The voice product logged the most activity during home hours.

